### ISSUE TITLE
Uncaught TypeError: Cannot read property 'length' of null on showPreview for invisible matrix with text processing

### ISSUE THREAD

**Issue by waynep16** [surveyjs/survey-library#3176]
DEMO 

https://survey.boomerangmessaging.com/test/#

### Are you requesting a feature, reporting a bug or asking a question?

BUG when completing 

### What is the current behavior?

When pressing the final complete button, the fields disable and there is no way to complete the form. The console log shows  Uncaught TypeError: Cannot read property 'length' of null error

### What is the expected behavior?

The form to  go to the overview screen.

### How would you reproduce the current behavior (if this is a bug)?

By taking the whole form

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

https://survey.boomerangmessaging.com/test/#

Test code

```
{
 title: "Lone Worker Demo Form",
 logoPosition: "none",
 completedHtml: "COMPLETED HTML HERE",
 pages: [
  {
   name: "On boarding",
   elements: [
    {
     type: "html",
     name: "start",
     html: "<img src=\"https://survey.boomerangmessaging.com/wp-content/uploads/2021/06/Artboard-1-copy-15.jpg\" />"
    }
   ],
   title: "On boarding"
  },
  {
   name: "Activation",
   elements: [
    {
     type: "imagepicker",
     name: "lone_worker_type",
     title: "Select a Lone Worker Type",
     valueName: "lone_worker_types",
     isRequired: true,
     requiredErrorText: "Select a lone worker type",
     choices: [
      {
       value: "public",
       text: "Lone Worker Public",
       imageLink: "https://survey.boomerangmessaging.com/wp-content/uploads/2021/06/Low-signal-icon-copy.png"
      },
      {
       value: "private",
       text: "Lone Worker Closed User Group",
       imageLink: "https://survey.boomerangmessaging.com/wp-content/uploads/2021/06/Maintains-personal-privacy-copy.png"
      }
     ],
     colCount: 2,
     imageHeight: 80,
     imageWidth: 80,
     showLabel: true
    },
    {
     type: "panel",
     name: "trigger_sms",
     elements: [
      {
       type: "dropdown",
       name: "activation_number",
       width: "49%",
       minWidth: "200px",
       maxWidth: "",
       title: "Choose Activation Number",
       hideNumber: true,
       isRequired: true,
       requiredErrorText: "Select an activation number",
       choices: [
        "0776459898",
        "0775198964",
        "0787464879"
       ],
       optionsCaption: "Select number"
      },
      {
       type: "text",
       name: "activation_keyword_sms",
       width: "49%",
       minWidth: "200px",
       maxWidth: "",
       startWithNewLine: false,
       title: "Activation Keyword",
       hideNumber: true,
       isRequired: true,
       requiredErrorText: "Specify an activation keyword",
       placeHolder: "Keyword",
       dataList: [
        "yes",
        "go",
        "start",
        "activate",
        "start"
       ]
      }
     ],
     visibleIf: "{lone_worker_types} notempty",
     title: "SMS Activation Trigger",
     description: "DESCRIPTION HERE NEEDED",
     isRequired: true,
     requiredErrorText: "Please configure your SMS activation",
     showNumber: true
    }
   ],
   title: "Lone Worker Activation",
   maxTimeToFinish: 2
  },
  {
   name: "Registration",
   elements: [
    {
     type: "html",
     name: "Activation Options",
     html: "<div class=\"sv_q_erbox\" style=\"border: 1px solid #007cba;  background-color: #dfedfc; color: #007cba;\">You have chosen <b>{lone_worker_types}</b> to be sent from <b>{activation_number}</b> with the activation keyphrase <b>{activation_keyword_sms}</b></div>"
    },
    {
     type: "comment",
     name: "registration_message",
     title: "Registration Message",
     description: "Message sent to a Lone Worker upon registration.  They will be asked to respond to this message to activate the lone Worker Session.",
     defaultValue: "Thank you for registering to [[customer_name]]’s lone worker monitoring service. Please reply to this message with your location to activate the lone worker session.  You will receive the first monitoring message immediately after replying to this message and every [[interval]] mins thereafter.",
     isRequired: true,
     requiredErrorText: "You need to specify a registration message",
     placeHolder: "Enter a custom registration message"
    },
    {
     type: "html",
     name: "Reg message info",
     startWithNewLine: false,
     hideNumber: true,
     html: "<h5 style=\"margin-bottom:5px\">Available Tokens</h5><a href=\"#\" class=\"reset button\" style=\"padding: 10px;float: right;\">Reset Text</a>\n<ul>\n<li>[[customer_name]]</li>\n<li>[[interval]]</li>\n</ul>\n\n"
    },
    {
     type: "boolean",
     name: "registration_repeat",
     minWidth: "",
     maxWidth: "",
     title: "Repeat Message If No Response? ",
     description: "If the user fails to respond, we will repeat your message",
     isRequired: true,
     showTitle: true
    },
    {
     type: "matrixdropdown",
     name: "repeat_options",
     useDisplayValuesInTitle: false,
     visibleIf: "{registration_repeat} = true",
     minWidth: "",
     maxWidth: "",
     title: "Repeat Options",
     description: "This wait time will also be applied to the monitoring interval.",
     hideNumber: true,
     defaultValue: {
      "Repeat Options": {
       Repeat: 2,
       Times: 5
      },
      the_repeat_options: {
       option_repeat: 2,
       option_time: 5
      }
     },
     isRequired: true,
     requiredErrorText: "Repeat options are required",
     columns: [
      {
       name: "option_repeat",
       title: "Repeat X message times",
       cellType: "rating",
       isRequired: true
      },
      {
       name: "option_time",
       title: "Wait X mins between messages",
       cellType: "text",
       isRequired: true,
       validators: [
        {
         type: "numeric",
         minValue: 5
        }
       ],
       inputType: "number",
       step: 5
      }
     ],
     rows: [
      {
       value: "the_repeat_options",
       text: "   "
      }
     ]
    },
    {
     type: "boolean",
     name: "regitration_failure",
     minWidth: "",
     title: "Send Registration Failure Message?",
     description: "Send a notification to the Lone Worker telling them that they failed to respond in time",
     defaultValue: "true",
     isRequired: true,
     readOnly: true
    },
    {
     type: "comment",
     name: "regitration_failure_message",
     visibleIf: "{regitration_failure} = true",
     startWithNewLine: false,
     title: "Registration Failure Message",
     hideNumber: true,
     defaultValue: "As you failed to respond within the required period, you are NOT registered for lone worker monitoring. If you still wish to register, please start the process again.",
     isRequired: true,
     requiredErrorText: "Please specify a failure message",
     placeHolder: "Registration Failure Message"
    },
    {
     type: "boolean",
     name: "registration_notification",
     minWidth: "",
     title: "Notify System User(s) On Activation?",
     description: "Send a notification when the Lone Worker process is activated.",
     isRequired: true,
     requiredErrorText: "Do you want to notify system users?"
    },
    {
     type: "html",
     name: "registration_notification_contacts",
     visibleIf: "{registration_notification} = true",
     minWidth: "",
     html: "<h5>CONTACT SELECTION PANEL HERE</h5>"
    },
    {
     type: "comment",
     name: "registration_notification_message",
     visibleIf: "{registration_notification} = true",
     title: "Notification Message",
     description: "Description here and indication about values?  Notification will be sent via SMS",
     defaultValue: "The Lone worker monitoring service has been activated by [name] / [[number]] at [[date]] / [[time]]. The location provided is [[reply_content]]",
     isRequired: true,
     placeHolder: "Enter your notification content"
    }
   ],
   title: "Registration"
  },
  {
   name: "Monitoring",
   elements: [
    {
     type: "html",
     name: "Reg settings",
     visible: false,
     html: "1.{repeat_options.option_repeat} <br>\n2.{repeat_option.option_repeat}<br>\n3.{option_repeat.wayne}<br>\n4.{the_repeat_options.option_repeat} <br>\n5.{wayne.option_repeat}<br>\n6.{the_repeat_options.option_repeat}<br>\n6.{the_repeat_option.option_repeat}<br>\n7.{repeat_options.wayne.option_repeat}<br>\n8.{repeat_options.wayne}<br>\n9.{wayne.option_repeat}<br>\n10 --  {repeat_options.the_repeat_options.option_repeat}"
    },
    {
     type: "comment",
     name: "monitoring_content",
     title: "Monitoring Message Content",
     description: "After registration, the monitoring process will begin. Send the below message at your set intervals to your Lone Worker.",
     defaultValue: "This is [[customer_name]]’s lone worker monitoring service. Please reply with one of the following numbers:\n\n1 – I’m safe\n9 – I need urgent assistance\n0 – Log off from lone worker monitoring service\n8 – My location has changed (provide your new location)\n\nIf you fail to respond to this message, [[no_of_repeats]] further message(s) will be sent at [[interval]] mins intervals. If you do not reply with [[interval]] mins of the final message, a lone working incident will be created.",
     rows: 10,
     placeHolder: "Monitoring Message Content"
    },
    {
     type: "html",
     name: "monitoring_help",
     startWithNewLine: false,
     hideNumber: true,
     html: "<h5 style=\"margin-bottom:5px\">Available Tokens</h5><a href=\"#\" class=\"reset button\" style=\"padding: 10px;float: right;\">Reset Text</a>\n<ul>\n<li>[[customer_name]]</li>\n<li>[[interval]]</li>\n<li>[[no_of_repeats]]</li>\n</ul>\n\n"
    },
    {
     type: "dropdown",
     name: "monitoring_frequncy",
     title: "Monitoring Frequency",
     description: "Specify how often you want to check on your Lone Worker",
     valueName: "monitoring_frequency",
     isRequired: true,
     requiredErrorText: "Monitoring Frequency is required",
     choices: [
      {
       value: "30",
       text: "Every 30 mins"
      },
      {
       value: "60",
       text: "Every 1 Hour"
      },
      {
       value: "120",
       text: "Every 2 Hours"
      },
      {
       value: "180",
       text: "Every 3 Hours"
      },
      {
       value: "240",
       text: "Every 4 Hours"
      },
      {
       value: "300",
       text: "Every 5 Hours"
      }
     ],
     hasOther: true,
     otherPlaceHolder: "Time in minutes",
     otherText: "Custom",
     otherErrorText: "Specify a custom time period",
     optionsCaption: "Monitoring Frequency"
    },
    {
     type: "boolean",
     name: "repeat_monitoring_message",
     title: "Repeat Monitoring Message If No Response?",
     isRequired: true
    },
    {
     type: "panel",
     name: "monitor_repeat",
     elements: [
      {
       type: "rating",
       name: "monitor_repeat_time",
       width: "50%",
       minWidth: "",
       startWithNewLine: false,
       title: "Repeat Message",
       description: "Will repeat {monitor_repeat} times",
       hideNumber: true,
       valueName: "monitor_repeat",
       defaultValue: 2,
       isRequired: true,
       requiredErrorText: "Repeat times required"
      },
      {
       type: "dropdown",
       name: "monitor_repeat_between",
       width: "50%",
       minWidth: "",
       startWithNewLine: false,
       title: "Time Between Messages",
       description: "Wait {monitor_time}",
       hideNumber: true,
       valueName: "monitor_time",
       defaultValue: "1",
       isRequired: true,
       requiredErrorText: "Time between messages is required",
       choices: [
        {
         value: "1",
         text: "5 Minutes"
        },
        {
         value: "2",
         text: "10 Minutes"
        },
        {
         value: "3",
         text: "15 Minutes"
        },
        {
         value: "4",
         text: "30 Minutes"
        },
        {
         value: "5",
         text: "45 Minutes"
        },
        {
         value: "6",
         text: "1 Hour"
        },
        {
         value: "7",
         text: "Custom"
        }
       ]
      }
     ],
     visibleIf: "{repeat_monitoring_message} = true",
     title: "Monitoring Repeat Options",
     startWithNewLine: false,
     width: "100%"
    }
   ],
   title: "Monitoring"
  },
  {
   name: "Escalation",
   elements: [
    {
     type: "html",
     name: "management_info",
     html: "<div class=\"sv_q_erbox\" style=\"border: 1px solid #007cba;  background-color: #dfedfc; color: #007cba;\">Monitoring messages will be sent every<b>{monitoring_frequency}</b> minutes.</div>"
    },
    {
     type: "comment",
     name: "escalation_message",
     useDisplayValuesInTitle: false,
     width: "70%",
     minWidth: "",
     title: "Incident Escalation Message",
     description: "Send a message if the Lone Worker Fails to check in to monitoring messages",
     defaultValue: "Lone worker [[name]] / [[number]], [[ref]], has failed to respond to the latest monitoring message sent on [[send_date]] / [[send_time]]. The last response received was at [[response_date]]  / [[response_time]] and the last location provided was [[location]]. Reply with:\n\n1 – The incident is being attended to – On scene\n2 – I will attend to the incident + ETA No. in Mins\n9 – Request back-up / assistance (Provide description of help required)\n0 – Lone worker is safe + Provide description of the situation / action taken – Close incident",
     isRequired: true,
     requiredErrorText: "You need to supply an escalation message",
     rows: 14,
     placeHolder: "Escalation Message"
    },
    {
     type: "html",
     name: "Escalation Tags",
     width: "30%",
     minWidth: "",
     startWithNewLine: false,
     html: "<h5 style=\"margin-bottom:5px\">Available Tokens</h5>\n<ul>\n<li>[[name]]</li>\n<li>[[number]]</li>\n<li>[[ref]]</li>\n<li>[[send_date]]</li>\n<li>[[send_time]]</li>\n<li>[[response_date]]</li>\n<li>[[response_time]]</li>\n<li>[[location]]</li>\n</ul>\n\n\n"
    },
    {
     type: "checkbox",
     name: "escalate_channel",
     width: "100%",
     startWithNewLine: false,
     title: "Send Escalation Message Via",
     hideNumber: true,
     defaultValue: [
      "1"
     ],
     isRequired: true,
     requiredErrorText: "Escalation comms channel is required",
     choices: [
      {
       value: "1",
       text: "SMS"
      },
      {
       value: "2",
       text: "Email"
      },
      {
       value: "3",
       text: "Voice"
      }
     ],
     colCount: 4,
     hasSelectAll: true
    },
    {
     type: "dropdown",
     name: "escalate_time",
     title: "How Long To Wait Before Sending?",
     description: "If there is no response, wait this amount of time before escalating. This will be sent after the final monitor repeat message of {monitor_time}.",
     isRequired: true,
     requiredErrorText: "You need to specify a wait time",
     choices: [
      {
       value: "5",
       text: "5 Minutes"
      },
      {
       value: "10",
       text: "10 Minutes"
      },
      {
       value: "15",
       text: "15 Minutes"
      },
      {
       value: "20",
       text: "20 Minutes"
      },
      {
       value: "30",
       text: "30 Minutes"
      },
      {
       value: "40",
       text: "40 Minutes"
      },
      {
       value: "50",
       text: "50 Minutes"
      },
      {
       value: "60",
       text: "1 Hour"
      }
     ],
     hasOther: true,
     otherPlaceHolder: "Time in minutes",
     otherText: "Custom",
     otherErrorText: "Enter time in minutes",
     optionsCaption: "Choose a time"
    },
    {
     type: "checkbox",
     name: "escalate_notifications",
     title: "Who Should Be Notified?",
     defaultValue: [
      "person"
     ],
     choices: [
      {
       value: "person",
       text: "Lone Worker"
      },
      {
       value: "contact",
       text: "A Contact"
      },
      {
       value: "group",
       text: "A Group"
      },
      {
       value: "custom",
       text: "Custom User"
      }
     ],
     colCount: 4,
     selectAllText: "All"
    },
    {
     type: "html",
     name: "escalate_group",
     visibleIf: "{escalate_notifications} contains 'group'",
     html: "Group Selection HTML"
    },
    {
     type: "comment",
     name: "escalate_custom",
     visibleIf: "{escalate_notifications} contains 'custom'",
     title: "Custom Notification Entries",
     description: "Phone number or email addresses. One per line.",
     hideNumber: true,
     requiredErrorText: "Please specify custom users to be notified",
     placeHolder: "Phone number or email address"
    },
    {
     type: "comment",
     name: "escalation_update",
     title: "Escalation Update Message",
     description: "Description NEEDED here of what this is?",
     defaultValue: "An incident handler has responded to Lone worker incident [[ref]] regarding [[name]] / [[number]]. Response details:\n\n[[response_content]].",
     isRequired: true,
     requiredErrorText: "Escalation update message is required",
     placeHolder: "Escalation update message"
    },
    {
     type: "boolean",
     name: "escalate_stand_down",
     title: "Add A Stand Down Message?",
     description: "If the user responds during the escalation process, send a message?",
     isRequired: true,
     requiredErrorText: "Do you want to add a stand down message?"
    },
    {
     type: "comment",
     name: "escalate_stand_down_content",
     visibleIf: "{escalate_stand_down} = true",
     title: "Stand Down Message",
     description: "Description NEEDED here of what this is?",
     defaultValue: "Lone Worker [[name]] / [[number]], [[ref]] has now responded to confirm they are safe.  Anyone attending to this incident can now stand down.",
     isRequired: true,
     requiredErrorText: "The stand down message is required",
     placeHolder: "Escalation update message"
    }
   ],
   title: "Incident Management",
   navigationTitle: "Incident Management "
  }
 ],
 sendResultOnPageNext: true,
 showTitle: false,
 showPageTitles: false,
 showQuestionNumbers: "onPage",
 showProgressBar: "top",
 progressBarType: "buttons",
 previewText: "Confirm Settings",
 firstPageIsStarted: true,
 showPreviewBeforeComplete: "showAnsweredQuestions"
}

```

### Specify your

* browser: Chrome
* browser version: Latest
* surveyjs platform (angular or react or jquery or knockout or vue):
* surveyjs version: Latest

**Comment by andrewtelnov** [comment]
@waynep16 I was able to reproduce it. We will take a look.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@waynep16 Fixed by the commit above. We will include the fix into v1.8.60

Thank you,
Andrew
